src/gtc/unstructured/usid2_codegen.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+#
+# Eve Toolchain - GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2020, CSCS - Swiss National Supercomputing Center, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from eve import codegen
+from eve.codegen import FormatTemplate as as_fmt
+from eve.codegen import MakoTemplate as as_mako
+from gtc.unstructured import usid2
+
+
+class _KernelCallGenerator(codegen.TemplatedGenerator):
+    Sid = as_fmt("{name}")
+    SparseField = as_fmt("sid::rename_dimensions<dim::s, {connectivity}_tag>({name})")
+    Composite = as_mako(
+        "sid::composite::make<${', '.join(f'{i.name}_tag' for i in _this_node.items)}>(${', '.join(items)})"
+    )
+    Kernel = as_mako(
+        "call_kernel<${id_}>(d.${location_type}${''.join(f', {c}' for c in [primary] + secondaries)});"
+    )
+
+
+class _Generator(codegen.TemplatedGenerator):
+    def visit_Computation(self, node: usid2.Computation, **kwargs):
+        return self.generic_visit(
+            node, kernel_calls=[_KernelCallGenerator.apply(k) for k in node.kernels], **kwargs
+        )
+
+    Literal = as_mako("${dtype}{${value}}")
+    BinaryOp = as_fmt("({left} {op} {right})")
+    FieldAccess = as_fmt("field<{name}_tag>({location})")
+    VarAccess = as_fmt("{name}")
+    NeighborReduction = as_mako(
+        "${op}_neighbors<${dtype}, ${connectivity}_tag, ${max_neighbors}, ${has_skip_values.lower()}>"
+        + "([](auto &&${primary}, auto &&${secondary}) { return ${body}; }, ${primary}, strides, ${secondary})"
+    )
+    VarDecl = as_fmt("auto&& {name} = {init};")
+    Assign = as_fmt("{left} = {right};")
+    Kernel = as_mako(
+        """struct ${id_} {
+    GT_FUNCTION auto operator()() const {
+        return [](auto && ${_this_node.primary.name}, auto &&strides${''.join(f', auto&& {s.name}' for s in _this_node.secondaries)}) {
+            ${'\\n            '.join(body)}
```

Comment:
We have clang-format in the pipeline. No need to do manual formatting in the cpp generators.

Reply:
Done.